data/service/data-trigger.js

```diff
@@ -322,46 +319,76 @@ exports.DataTrigger.prototype = Object.create(
 
                     // Start an asynchronous fetch of the property's value if necessary.
                     this.getObjectProperty(object).catch((error) => {
-                        console.error(this._propertyName + " DataTrigger: getObjectProperty error: ", error);
+                        console.error(propertyName + " DataTrigger: getObjectProperty error: ", error);
                     });
                 }
 
                 //}
 
-                /**
-                 * We're going to initialize to-Many to their collection type.
-                 * I don't think we need to check our valueStatus to see if
-                 * a fetch is in flight or have been made, since we merge local value
-                 * and fetched ones in the setter.
-                 */
+                // Ensure to-Many properties are initialized to their collection type.
+                this._ensureCollectionValue(object);
+
+                // Return the property's current value.
+                return getter ? getter.call(object) : object[this._privatePropertyName];
+            },
+        },
+
+        /**
+         * @method
+         * @argument {Object} object
+         */
+        _ensureCollectionValue: {
+            value: function (object) {
+                const { isMandatory = false, collectionValueType } = this.propertyDescriptor;
+                const expectsCollection = this.isToMany || collectionValueType !== undefined;
+
+                if (!expectsCollection) return;
```

Comment:
This is a heavily trafficked piece of code, so performance is important. I found  out through perf testing that the cost of declaring a local variable is worth it if that variable is used more than 3 times. Here  -  expectsCollection - is only used once after being initialized, and I also find personally that for this use case it would be much more readable to have the actual logic in the if() so let's not create the expectsCollection variable